chalicelib/utils.py

```diff
@@ -127,23 +138,27 @@ def check_deco(func):
         def wrapper(*args, **kwargs):
             start_time = time.time()
             kwargs = handle_kwargs(kwargs, default_kwargs)
-            partials = {'name': func.__name__, 'kwargs': kwargs, 'is_check': True,
-                        'start_time': start_time, 'connection': args[0]}
-            signal.signal(signal.SIGALRM, partial(timeout_handler, partials))
-            signal.alarm(CHECK_TIMEOUT)  # run time allowed in seconds
-            try:
-                check = func(*args, **kwargs)
-                check.validate()
-            except Exception as e:
-                # connection should be the first (and only) positional arg
-                check = CheckResult(args[0], func.__name__)
-                check.status = 'ERROR'
-                check.description = 'Check failed to run. See full output.'
-                check.full_output = traceback.format_exc().split('\n')
-            signal.alarm(0)
-            kwargs['runtime_seconds'] = round(time.time() - start_time, 2)
-            check.kwargs = kwargs
-            return check.store_result()
+            parent_pid = os.getpid()
+            child_pid = os.fork()
```

Comment:
While a subprocess seems reasonable, I'm curious about the choice of `os.fork` vs some sort of `subprocess` choice.